src/python/joint_cell_plotting.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+"""
+This script QCs barcodes via ATAC frags & TSS and RNA UMIs & genes,
+and plots all barcodes colored by joint QC status. It also generates the 
+same plot with transparency added to show density. 
+"""
+
+import argparse
+import logging
+import numpy as np
+import pandas as pd
+from plotnine import *
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Plot barcodes by RNA and ATAC QC status")
+    parser.add_argument("pkr", help="PKR name")
+    parser.add_argument("rna_metrics_file", help="Filename for RNA metrics tsv matrix")
+    parser.add_argument("atac_metrics_file", help="Filename for ATAC metrics tsv file")
+    parser.add_argument("min_UMIs", type=int, help="Cutoff for minimum number of UMIs")
+    parser.add_argument("min_genes", type=int, help="Cutoff for minimum number of genes")
+    parser.add_argument("min_TSS", type=int, help="Cutoff for minimum TSS score")
+    parser.add_argument("min_frags", type=int, help="Cutoff for minimum number of ATAC fragments")
+    parser.add_argument("barcode_metadata_file", help="Filename for barcode metadata csv file")
+    
+    return parser.parse_args()
+
+def get_split_lines(file_name, delimiter):
+    with open(file_name, "r") as f:
+        for line in f:
+            yield line.rstrip().split(sep=delimiter)
+
+def merge_dicts(dict_1, dict_2):
+    """Merge dictionaries by key; combine values into quadruple, fill with 0s if key not in both dicts"""
+    keys = set(dict_1.keys() | dict_2.keys())
+    merged = {k: (dict_1.get(k, (0,0)) + dict_2.get(k, (0,0))) for k in keys}
+    
+    return(merged)
+
+def get_metrics(rna_metrics_file, atac_metrics_file):
+    """Read files and aggregate metrics into Pandas dataframe"""
+    rna_metrics_contents = get_split_lines(rna_metrics_file, delimiter="\t")
+    UMIs = []
+    genes = []
+    rna_barcodes = []
+    next(rna_metrics_contents) # skip header
```

Comment:
since you like using the 'get_split_line' function, you should add a parameter 'skip_header' so you don't have to do this next(line) here